app/services/project_data_service.rb

```diff
@@ -52,8 +54,31 @@ def update_dependencies
   end
 
   def update_dependency(name, version)
+    update_required = dependency_update_required?(name, version)
     @project.dependencies.find_or_initialize_by(name: name)
-            .update!(version: version)
+            .update!(version: version, update_required: update_required)
+  end
+
+  def dependency_update_required?(name, version)
```

Comment:
This is an interesting method. Unless I'm mistaken, I believe that it does NOT use the state on this class, correct? Like, I'm not seeing a direct relationship between the `project` that this service is instantiated with and the logic of this method. That leads me to believe it might be in the wrong location.

The arguments that are passed in are properties of a `dependency`, correct? If so, then I would expect to see this defined over there:

```rb
test_project = Project.create(name: "test", ...)
ruby_dep = test_project.dependencies.create(name: "ruby", version: "2.7.1", ...)
ruby_dep.update_required?
```

And then that `update_required?` method would take no args because the name and version are on the dependency instance. It could reach into the `Horizon.config` to get required versions and return the boolean.

Reply:
Moved this and some additional logic into models, and added `as_json`